core/arch/arm/plat-stm32mp1/drivers/stm32mp1_syscfg.c

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2019-2020, STMicroelectronics
+ */
+
+#include <dt-bindings/clock/stm32mp1-clks.h>
+#include <initcall.h>
+#include <kernel/delay.h>
+#include <mm/core_memprot.h>
+#include <stm32_util.h>
+#include <io.h>
+#include <trace.h>
+#include <types_ext.h>
+
+/*
+ * SYSCFG register offsets (base relative)
+ */
+#define SYSCFG_CMPCR				0x20U
+#define SYSCFG_CMPENSETR			0x24U
+
+/*
+ * SYSCFG_CMPCR Register
+ */
+#define SYSCFG_CMPCR_SW_CTRL			BIT(1)
+#define SYSCFG_CMPCR_READY			BIT(8)
+#define SYSCFG_CMPCR_RANSRC			GENMASK_32(19, 16)
+#define SYSCFG_CMPCR_RANSRC_SHIFT		16
+#define SYSCFG_CMPCR_RAPSRC			GENMASK_32(23, 20)
+#define SYSCFG_CMPCR_ANSRC_SHIFT		24
+
+#define SYSCFG_CMPCR_READY_TIMEOUT_US		1000U
+
+/*
+ * SYSCFG_CMPENSETR Register
+ */
+#define SYSCFG_CMPENSETR_MPU_EN			BIT(0)
+
+static vaddr_t get_syscfg_base(void)
+{
+	struct io_pa_va base = { .pa = SYSCFG_BASE };
+
+	return io_pa_or_va(&base);
+}
+
+void stm32mp_syscfg_enable_io_compensation(void)
+{
+	uintptr_t syscfg_base = get_syscfg_base();
```

Comment:
I'd prefer `vaddr_t`, same in `stm32mp_syscfg_disable_io_compensation()` below.

Reply:
ok, will do.